system/security/securesocket/securesocket.cpp

```diff
@@ -317,7 +318,26 @@ class CSecureSocket : implements ISecureSocket, public CInterface
 
     virtual unsigned __int64 getStatistic(StatisticKind kind) const override
     {
-        return m_socket->getStatistic(kind);
+        switch (kind)
```

Comment:
this could be moved as a shared function - either a member function of SocketStats, (or a global function which takes a SocketStats parameter) to avoid duplicating the code.

Reply:
I agree and that was one of my initial thoughts about why I did it the first way.
I will create a common function for this.